lib/pubsuffix-psl.ts

```diff
@@ -84,5 +84,5 @@ export function getPublicSuffix(
     )
   }
 
-  return psl.get(domain)
+  return getDomain(domain)
```

Comment:
Note: by default `tldts` ignores the private section of the public suffix list. If the intent here is to also take into account these rules (which I assume might be the case since the `tough-cookie` library deals with cookies and browsers take into account the private section of the public suffix list) then you'd need to add the following option while calling `getDomain`:
```suggestion
  return getDomain(domain, { allowPrivateDomains: true })
```

Additionally, `tldts` is able to handle arbitrary inputs (i.e. both URLs and hostnames alike) with a slight overhead for the detection. If the input is already a valid hostname (e.g. if it was extracted using a `URL` parser) you can squeeze a bit of extra performance using the `{ extractHostname: false }` option, which skips the logic of extracting hostnames  from URL inputs as well as hostname validation.

Lastly, `tldts` is also able to handle IPv4 and IPv6 inputs by default, which I believe `psl` does not support. So if these inputs are already handled in a different way you can also disable this verification in `tldts` using the option: `{ detectIp: false }` (which has some small overhead).

Reply:
Thanks for all those tips!